server/src/pages/shot/editor.js

```diff
@@ -34,23 +51,278 @@ exports.Editor = class Editor extends React.Component {
       </div>
       <div className="main-container inverse-color-scheme">
         <div className="canvas-container" id="canvas-container" ref={(canvasContainer) => this.canvasContainer = canvasContainer}>
-          <canvas className="image-holder centered" id="image-holder" ref={(image) => { this.imageCanvas = image }} height={ canvasHeight } width={ canvasWidth } style={{height: canvasHeight, width: canvasWidth}}></canvas>
-          <canvas className="highlighter centered" id="highlighter" ref={(highlighter) => { this.highlighter = highlighter }} height={canvasHeight} width={canvasWidth}></canvas>
-          <canvas className={"editor centered " + this.state.tool} id="editor" ref={(editor) => { this.editor = editor }} height={canvasHeight} width={canvasWidth}></canvas>
+          <canvas className="image-holder centered" id="image-holder" ref={(image) => { this.imageCanvas = image }} height={ this.canvasHeight } width={ this.canvasWidth } style={{height: this.canvasHeight, width: this.canvasWidth}}></canvas>
+          <canvas className="highlighter centered" id="highlighter" ref={(highlighter) => { this.highlighter = highlighter }} height={this.canvasHeight} width={this.canvasWidth}></canvas>
+          <canvas className="editor centered" id="editor" ref={(editor) => { this.editor = editor }} height={this.canvasHeight} width={this.canvasWidth}></canvas>
+          <canvas className="crop-tool centered" id="crop-tool" ref={(cropper) => { this.cropper = cropper }} height={this.canvasHeight} width={this.canvasWidth}></canvas>
         </div>
       </div>
     </div>
   }
 
   componentDidUpdate() {
+    if (this.state.tool != 'crop') {
+      this.cropToolBar = null;
+      this.cropper.removeEventListener("mousemove", this.mousemove);
+      this.cropper.removeEventListener("mouseout", this.mouseout);
+      this.cropper.removeEventListener("mousedown", this.mousedown);
+      this.cropper.removeEventListener("mouseup", this.mouseup);
+    }
+    this.edit();
+  }
+
+  onClickCrop() {
+    this.setState({tool: 'crop'});
+    this.cropToolBar = <div className="annotation-tools">
+      <button className={`button transparent confirm-crop`} id="confirm-crop" onClick={this.onClickConfirmCrop.bind(this)} title="confirm-crop">Confirm</button>
+      <button className={`button transparent cancel-crop`} id="cancel-crop" onClick={this.onClickCancelCrop.bind(this)} title="cancel-crop">Cancel</button>
+    </div>
+  }
+
+  onClickConfirmCrop() {
+    let x1 = boxes[0].x1;
+    let x2 = boxes[0].x2;
+    let y1 = boxes[0].y1;
+    let y2 = boxes[0].y2;
+    x1 = x1 < 0 ? 0 : x1;
+    y1 = y1 < 0 ? 0 : y1;
+    x2 = x2 > this.canvasWidth ? this.canvasWidth : x2;
+    y2 = y2 > this.canvasHeight ? this.canvasHeight : y2;
+    let cropWidth = x2 - x1;
+    let cropHeight = y2 - y1;
+    let croppedImage = document.createElement('canvas');
+    croppedImage.width = parseInt(cropWidth, 10);
+    croppedImage.height = parseInt(cropHeight, 10);
```

Comment:
I don't think `parseInt`s are needed.  But if they are, then the results should be assigned into `cropWidth` and `cropHeight` since those are assigned to other variables later.